addons/account/models/account_invoice.py

```diff
@@ -40,6 +39,18 @@
 
 MAGIC_COLUMNS = ('id', 'create_uid', 'create_date', 'write_uid', 'write_date')
 
+def calc_check_digits(number):
+    """Calculate the extra digits that should be appended to the number to make it a valid number.
+    Source: python-stdnum iso7064.mod_97_10.calc_check_digits
+    """
+    try:
+        number_base10 = ''.join(str(int(x, 36)) for x in number)
+    except Exception:
+        raise UserError(_("Invalid format for %s") % number)
```

Comment:
The old code didn't convert from internal stdnum exceptions to Odoo exceptions so maybe we can just let the ValueError bubble up if a non-alphanumeric IBAN is provided?

Reply:
right, removing the try/except